### ISSUE TITLE
I2C transmit function return value error, and add timeout support

### ISSUE THREAD

**Issue by Neutree** [kendryte/kendryte-standalone-sdk#38]
- [x] BUG REPORT
- [ ] FEATURE REQUEST

-------------------------------------------------------------------------------

### Test code

Pin28 and Pin29 didn't connected with any device, and run:

```c
#include <stdio.h>
#include "bsp.h"
#include "i2c.h"
#include "fpioa.h"
#include "sleep.h"
#include "sysctl.h"

uint8_t addr = 0x26; //Invalid  address, connected with no any slave device
int main()
{
    const uint8_t data[] = {1,2,3,4,5};
    plic_init();
    sysctl_enable_irq();
    fpioa_set_function(28,FUNC_I2C0_SCLK);
    fpioa_set_function(29,FUNC_I2C0_SDA);
    i2c_init(I2C_DEVICE_0, addr, 7, 100000);

    while(1)
    {
        int ret = i2c_send_data(I2C_DEVICE_0, data, sizeof(data));
        printf("ret:%d\n",ret);
        sleep(2);
    }
}
```

### Expected behavior

Cause I2C pin didn't connected with any device, so we should get some error hint like this:

```
ret:1
ret:1
ret:1
ret:1
.
.
.
```

### Actual behavior

But we get this result:
```
ret:0
ret:1
ret:1
ret:1
.
.
.
```
![_ _20181228141202](https://user-images.githubusercontent.com/8625829/50504735-9ca09b00-0aaa-11e9-93f2-9414f119b24c.png)

And just once wave we can see in logic analyzer, maybe the `Tx abort src` register not clean.

### Document version

latest lib with commit `c7148b55e95bb9bb1444413f49cbaf85d4e0c252`

### Hardware

Sipeed Maix Go/ Maix One

### System

Linux18.04 with toolchain 8.2

-------------------------------------------------------------------------------

**Comment by Neutree** [comment]
And if we send 2 Bytes data every time,
```
uint8_t data[2] = {0x00, 0xAE};
uint8_t data2[2] = {0x00, 0x20};
 i2c_send_data(I2C_DEVICE_0, data, 2);
 i2c_send_data(I2C_DEVICE_0, data2, 2);
```
send twice but only get one stop signal:

![_ _20181228165649](https://user-images.githubusercontent.com/8625829/50509666-c1ecd380-0ac1-11e9-97c9-091631ad7590.png)

**Comment by Neutree** [comment]
I changed wait condition, it works now!

```
    while ( (i2c_adapter->status & I2C_STATUS_ACTIVITY) ||
            !(i2c_adapter->status & I2C_STATUS_TFE) )
        ;
```

**Comment by zzxcanaan** [comment]
Thanks, we will fix it in version 0.5.3
